src/main/java/net/minecraftforge/client/GeneratorTypeManager.java

```diff
@@ -0,0 +1,196 @@
+     * @return The GeneratorType at the given position
+     * @throws IndexOutOfBoundsException if the ordinal is out of range
+     */
+    public synchronized BiomeGeneratorTypeScreens getGeneratorTypeAt(int index) {
+        return generatorTypes.get(index);
+    }
+
+    @Nonnull
+    public synchronized BiomeGeneratorTypeScreens getDefaultGeneratorType() {
+        return defaultGeneratorType;
+    }
+
+    /**
+     * Get the EditScreenFactory registered to the given GeneratorType
+     *
+     * @param generatorType The GeneratorType to get an EditScreenFactory for
+     * @return The EditScreenFactory or null
+     */
+    @Nullable
+    public synchronized BiomeGeneratorTypeScreens.IFactory getEditScreenFactory(BiomeGeneratorTypeScreens generatorType) {
+        return editScreens.get(generatorType);
+    }
+
+    /**
+     * Register a GeneratorType with the GeneratorTypeManager
+     *
+     * @param generatorType The GeneratorType to be added
+     * @return true if the GeneratorType was successfully added
+     */
+    public synchronized boolean register(BiomeGeneratorTypeScreens generatorType) {
+        if (!generatorTypes.contains(generatorType)) {
+            generatorTypes.add(generatorType);
+            return true;
+        }
+        return false;
+    }
+
+    /**
+     * Register a GeneratorType and associated EditScreenFactory with the GeneratorTypeManager
+     *
+     * @param generatorType The GeneratorType to be added
+     * @param editScreen    The EditScreenFactory to be associated with given GeneratorType
+     * @return true if the GeneratorType and EditScreenFactory was successfully added
+     */
+    public synchronized boolean register(BiomeGeneratorTypeScreens generatorType, BiomeGeneratorTypeScreens.IFactory editScreen) {
+        if (register(generatorType)) {
+            editScreens.put(generatorType, editScreen);
+            return true;
+        }
+        return false;
+    }
+
+    /**
+     * Set the EditScreenFactory for a given GeneratorType, overriding any previously set factory
+     *
+     * @param generatorType The GeneratorType to associate with the given EditScreenFactory
+     * @param editScreen    The EditScreenFactory to associate with the given GeneratorType
+     * @return true if the EditScreenFactory was successfully set for the GeneratorType
+     */
+    public synchronized boolean setEditScreen(BiomeGeneratorTypeScreens generatorType, BiomeGeneratorTypeScreens.IFactory editScreen) {
+        if (generatorTypes.contains(generatorType)) {
+            editScreens.put(generatorType, editScreen);
+            return true;
+        }
+        return false;
+    }
+
+    /**
+     * Set the default GeneratorType
+     *
+     * @param generatorType The GeneratorType to use as the default
+     */
+    public synchronized void setDefaultGeneratorType(@Nonnull BiomeGeneratorTypeScreens generatorType) {
+        this.defaultGeneratorType = generatorType;
+    }
```

Comment:
This is probably a bad idea in that a default is a single object, and it is in an environment with multiple actors.
This is nothing more than a race condition where multiple mods will be competing to be the loser to set the default.
The default should be hard-set to the vanilla default (`field_239066_a_`) only.

Reply:
I'd say its fine, if (for example) modpacks want to have a default value for some reason. I would perhaps just add logging after the first override.

Reply:
To second cyborg - there is no possible *breaking conflict* that would be caused by having two mods compete for a default. One would win, the other would be just fine. If anything, this avoids two mods doing the same mixin / coremod (of which I am doing in lieu of this PR) to do this which is going to be much worse if two mods do the same thing. This pattern exists all over (see: registry replacement, recipe / json overwriting, etc.)

And as to why this would be useful, for people writing complete world gen overhauls, I get a lot of bug reports and confused users by "I installed this mod that's supposed to completely change world gen but I got a vanilla world". This kind of default helps deal with that.

Reply:
Huh, thought I put note about about 'no guarantees' on the javadoc.

But yeah, this is something I've seen quite a few mods do with coremods below 1.16. Only one can win out in that situation too (assuming they haven't exploded trying).

Reply:
A second idea might be to say whichever entry is earliest in the list wins as default. That way the number of clicks to select the loser's option is minimized (were it the other way round the user would have to cycle back round through the options to get to a previous spot)

Reply:
Marking as resolved as this is now outdated. I've added the logging as suggested in the changes